File: schemas/project.py
from pydantic import BaseModel
from typing import Optional

class ProjectBase(BaseModel):
    project_name: str
    created_by_name: Optional[str] = None
    created_by_role: Optional[str] = None
    created_by_user_id: Optional[int] = None
    module_name: Optional[str] = None
    scenario_name: Optional[str] = None

class ProjectCreate(ProjectBase):
    pass

class Project(ProjectBase):
    project_id: int

    class Config:
        from_attributes = True

File: src/types/project.ts
export interface Project {
  Project_ID: number;
  Project_Name: string;
  Created_By_Name?: string;
  Created_By_Role?: string;
  Created_By_User_ID?: number;
  Module_Name?: string;
  Scenario_Name?: string;
}

export interface ProjectCreate {
  Project_Name: string;
  Created_By_Name?: string;
  Created_By_Role?: string;
  Created_By_User_ID?: number;
  Module_Name?: string;
  Scenario_Name?: string;
}
